bumble/controller.py

```diff
@@ -623,32 +994,42 @@ def on_classic_role_change(self, peer_address, new_role):
     # Advertising support
     ############################################################
     def on_advertising_timer_fired(self):
-        self.send_advertising_data()
-        self.advertising_timer_handle = asyncio.get_running_loop().call_later(
-            self.advertising_interval / 1000.0, self.on_advertising_timer_fired
-        )
+        self.advertising_timer_handle = None
 
-    def start_advertising(self):
-        # Stop any ongoing advertising before we start again
-        self.stop_advertising()
+        self.send_advertising_data()
 
-        # Advertise now
-        self.advertising_timer_handle = asyncio.get_running_loop().call_soon(
-            self.on_advertising_timer_fired
+        # Compute the time of the next advertisement
+        next_advertisement = min(
+            (
+                advertiser.next_advertising_time
+                for advertiser in itertools.chain(
+                    (self.legacy_advertiser,), self.extended_advertisers.values()
+                )
+                if advertiser.enabled
+            ),
+            default=0,
         )
 
-    def stop_advertising(self):
-        if self.advertising_timer_handle is not None:
-            self.advertising_timer_handle.cancel()
-            self.advertising_timer_handle = None
+        if next_advertisement:
+            # We have at least one advertiser
+            delay = max(next_advertisement - time.time(), 0)
+            self.advertising_timer_handle = asyncio.get_running_loop().call_later(
+                delay, self.on_advertising_timer_fired
+            )
+
+    def start_advertising_timer(self):
+        if self.advertising_timer_handle is None:
+            self.advertising_timer_handle = asyncio.get_running_loop().call_soon(
+                self.on_advertising_timer_fired
+            )
 
     def send_advertising_data(self):
-        if self.link and self.advertising_data:
-            self.link.send_advertising_data(self.random_address, self.advertising_data)
+        # Legacy advertising
+        self.legacy_advertiser.send_advertising_data(self.link)
 
-    @property
-    def is_advertising(self):
-        return self.advertising_timer_handle is not None
+        # Extended advertising
```

Comment:
Should all advertisers send data on each of them was scheduled?

Reply:
This is refactored now